src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -745,6 +744,7 @@ public Task<RunspaceHandle> GetRunspaceHandleAsync(CancellationToken cancellatio
                 // Don't change our SessionState for ReadLine.
                 if (!executionOptions.IsReadLine)
                 {
+                    await this.sessionStateLock.WaitAsync();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5079709)

Reply:
Good point @TylerLeonhardt